packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1375,3 +1378,182 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
+      hre.network.name,
+      configPath
+    )
+  }
+
+  spinner.succeed('Project registration detected')
+  spinner.start('Checking proxy compatibility...')
+
+  const incompatibleProxyError = `ChugSplash does not support your proxy type.
+  Currently ChugSplash only supports proxies that implement EIP-1967 which yours does not appear to do.
+  If you believe this is a mistake, please reach out to the developers or open an issue on GitHub.`
+
+  // Fetch proxy bytecode and check if it contains the expected EIP-1967 function definitions
+  const iface = new ethers.utils.Interface(ProxyABI)
+  const bytecode = await provider.getCode(proxyAddress)
+  const checkFunctions = ['implementation', 'admin', 'upgradeTo', 'changeAdmin']
+  for (const func of checkFunctions) {
+    const sigHash = remove0x(iface.getSighash(func))
+    if (!bytecode.includes(sigHash)) {
+      throw new Error(incompatibleProxyError)
+    }
+  }
+
+  // Fetch proxy owner address from storage slot defined by EIP-1967
+  const ownerAddress = ethers.utils.defaultAbiCoder.decode(
+    ['address'],
+    await provider.getStorageAt(
+      proxyAddress,
+      '0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103'
+    )
+  )[0]
+
+  // Fetch ChugSplashManager address for this project
+  const managerAddress = getChugSplashManagerProxyAddress(
+    parsedConfig.options.projectName
+  )
+
+  // If proxy owner is not a valid address, then proxy type is incompatible
+  if (!ethers.utils.isAddress(ownerAddress)) {
+    throw new Error(incompatibleProxyError)
+  }
+
+  // If proxy owner is already ChugSplash, then throw an error
+  if (managerAddress.toLowerCase() === ownerAddress.toLowerCase()) {
+    throw new Error('Proxy is already owned by ChugSplash')
+  }
+
+  // If the signer doesn't own the target proxy, then throw an error
+  const signerAddress = await signer.getAddress()
+  if (ownerAddress.toLowerCase() !== signerAddress.toLowerCase()) {
+    throw new Error(`Target proxy is owned by: ${ownerAddress}.
+You attempted to transfer ownership of the proxy using the address: ${signerAddress}`)
+  }
+
+  // Check that the proxy implementation function returns an address
+  const contract = new ethers.Contract(proxyAddress, iface, signer)
+  const implementationAddress = await contract.callStatic.implementation()
+  if (!ethers.utils.isAddress(implementationAddress)) {
+    throw new Error(incompatibleProxyError)
+  }
+
+  spinner.succeed('Proxy compatibility verified')
+  spinner.start('Transferring proxy ownership to ChugSplash...')
+
+  // Transfer ownership of the proxy to the ChugSplashManager.
+  await (await contract.changeAdmin(managerAddress)).wait()
+
+  spinner.succeed('Proxy ownership successfully transferred to ChugSplash')
+}
+
+task(TASK_CHUGSPLASH_TRANSFER_OWNERSHIP)
+  .addPositionalParam(
+    'configPath',
+    'Path to the ChugSplash config file for the project that you would like to own the target contract'
+  )
+  .addPositionalParam(
```

Comment:
> Was just messing around with these tasks and realized that having two positional params is pretty unintuitive. E.g. running `npx hardhat chugsplash-claim-proxy <configPath> <referenceName>` works, but switching their order (i.e. `<referenceName> <configPath>`) doesn't. In all of the Hardhat tasks I've used, there's only one positional param to avoid this. What do you think about changing the reference name to `addFlag`?

Same thing here. We could change the flag to be `proxy` or keep it as `proxyAddress` if you'd prefer